backend/incident_reader.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+type IncidentDataReader struct {
+	db *sql.DB
+}
+
+func NewIncidentDataReader(db *sql.DB) *IncidentDataReader {
+	return &IncidentDataReader{db: db}
+}
+
+func (r *IncidentDataReader) ReadRelatedIncidents(ctx context.Context, incidentTitle string, monitorID string) ([]Incident, error) {
+	dbCon, err := r.db.Conn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := dbCon.Close()
+		if err != nil {
+			log.Warn().Stack().Err(err).Msg("Failed to close connection")
+		}
+	}()
+
+	rows, err := dbCon.QueryContext(ctx, "SELECT monitor_id, title, description, timestamp, severity, status FROM incident_data WHERE monitor_id = ? AND title = ? ORDER BY created_by DESC", monitorID, incidentTitle)
```

Comment:
Did you mean `created_at` instead of `created_by`?
```suggestion
	rows, err := dbCon.QueryContext(ctx, "SELECT monitor_id, title, description, timestamp, severity, status FROM incident_data WHERE monitor_id = ? AND title = ? ORDER BY created_at DESC", monitorID, incidentTitle)
```